tmt/steps/provision/mrack.py

```diff
@@ -1458,3 +1463,22 @@ def guest(self) -> Optional[GuestBeaker]:
         """
 
         return self._guest
+
+
+@container
+class GuestLogBeaker(tmt.steps.provision.GuestLog):
+    def __init__(self, guest: GuestBeaker, name: str, url: Optional[str] = None) -> None:
+        self.name = name
+        self.url = url
+        self.guest = guest
+
+    def fetch(self) -> Optional[str]:
+        """
+        Fetch and return content of a log.
+
+        :returns: content of the log, or ``None`` if the log cannot be retrieved.
+        """
+
+        if self.name == 'dmesg':
+            return self.guest.execute(Command('dmesg')).stdout
+        return tmt.utils.get_url_content(self.url) if self.url else None
```

Comment:
1 `dmesg` should have its own subclass, right? It has a different implementation of `fetch()` after all
2. I'm not sure it would be always possible to run `self.guest.execute()`, the guest may be dead. It would be better to download Beaker's `console.log` instead.